src/test/java/io/nitric/api/secret/SecretsTest.java

```diff
@@ -0,0 +1,272 @@
+
+import io.nitric.proto.secret.v1.SecretAccessRequest;
+import io.nitric.proto.secret.v1.SecretAccessResponse;
+import io.nitric.proto.secret.v1.SecretPutRequest;
+import io.nitric.proto.secret.v1.SecretPutResponse;
+import io.nitric.proto.secret.v1.SecretServiceGrpc;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SecretsTest {
+
+    @Test
+    public void test_serviceStub() {
+        assertNotNull(Secrets.getServiceStub());
+
+        var mock = Mockito.mock(SecretServiceGrpc.SecretServiceBlockingStub.class);
+
+        Secrets.setServiceStub(mock);
+        assertEquals(mock, Secrets.getServiceStub());
+    }
+
+    @Test
+    public void test_secret() {
+
+        var secret = Secrets.secret("name");
+        assertNotNull(secret);
+        assertEquals("name", secret.getName());
+        assertEquals("Secret[name=name]", secret.toString());
+
+        try {
+            Secrets.secret(" ");
+            fail();
+        } catch (IllegalArgumentException iae) {
+        }
+    }
+
+    @Test
+    public void test_secret_version() {
+        var secretVersion = Secrets.secret("name").version("version");
+        assertNotNull(secretVersion);
+        assertEquals("name", secretVersion.getSecret().getName());
+        assertEquals("version", secretVersion.getVersion());
+        assertEquals("SecretVersion[secret=Secret[name=name], version=version]", secretVersion.toString());
+
+        try {
+            Secrets.secret("name").version(" ");
+            fail();
+        } catch (IllegalArgumentException iae) {
+        }
+    }
+
+    @Test
+    public void test_secret_latest() {
+        var secretVersion = Secrets.secret("name").latest();
+        assertNotNull(secretVersion);
+        assertEquals("name", secretVersion.getSecret().getName());
+        assertEquals("latest", secretVersion.getVersion());
+        assertEquals("SecretVersion[secret=Secret[name=name], version=latest]", secretVersion.toString());
+    }
+
+    @Test
+    public void test_secret_put() {
+        var secret = io.nitric.proto.secret.v1.Secret.newBuilder().setName("name").build();
+        var secretVersion = io.nitric.proto.secret.v1.SecretVersion.newBuilder()
+                .setSecret(secret)
+                .setVersion("version")
+                .build();
+
+        var mock = Mockito.mock(SecretServiceGrpc.SecretServiceBlockingStub.class);
+        Mockito.when(mock.put(Mockito.any(SecretPutRequest.class))).thenReturn(
+                SecretPutResponse.newBuilder().setSecretVersion(secretVersion).build()
+        );
+        Secrets.setServiceStub(mock);
+
+        var value = "password".getBytes(StandardCharsets.UTF_8);
+
+        var sv = Secrets.secret("name").put(value);
+
+        assertNotNull(sv);
+        assertEquals("name", sv.getSecret().getName());
+        assertEquals("version", sv.getVersion());
+
+        // Verify we actually called the mock object
+        Mockito.verify(mock).put(Mockito.any());
```

Comment:
Would be good to verify that the payload the RPC client is being given in these cases. I've been trying to get out of the `any` matcher habit recently and have caught a few bugs.

Reply:
Done